src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -167,151 +267,52 @@ object ApiOperations {
-
-  private def filterOperationIDs(filter: Filter, og: Tables.OperationGroups): Rep[Boolean] =
-    if (filter.operationGroupIDs.isDefined && filter.operationGroupIDs.get.nonEmpty)
-      og.hash.inSet(filter.operationGroupIDs.get)
-    else true
-
-  private def filterOperationIDs(filter: Filter, o: Tables.Operations): Rep[Boolean] =
-    if (filter.operationGroupIDs.isDefined && filter.operationGroupIDs.get.nonEmpty)
-      o.operationGroupHash.inSet(filter.operationGroupIDs.get)
-    else true
-
-  private def filterOperationSources(filter: Filter, o: Tables.Operations): Rep[Boolean] =
-    if (filter.operationSources.isDefined && filter.operationSources.get.nonEmpty)
-      o.source.getOrElse("").inSet(filter.operationSources.get)
-    else true
-
-  private def filterOperationDestinations(filter: Filter, o: Tables.Operations): Rep[Boolean] =
-    if (filter.operationDestinations.isDefined && filter.operationDestinations.get.nonEmpty)
-      o.destination.getOrElse("").inSet(filter.operationDestinations.get)
-    else true
-
-  private def filterOperationParticipants(filter: Filter, o: Tables.Operations): Rep[Boolean] =
-    if (filter.operationParticipants.isDefined && filter.operationParticipants.get.nonEmpty) {
-      o.destination.getOrElse("").inSet(filter.operationParticipants.get) ||
-        o.source.getOrElse("").inSet(filter.operationParticipants.get)
-    }
-    else true
-
-  private def filterAccountIDs(filter: Filter, a: Tables.Accounts): Rep[Boolean] =
-    if (filter.accountIDs.isDefined && filter.accountIDs.get.nonEmpty) a.accountId.inSet(filter.accountIDs.get) else true
-
-  private def filterAccountManagers(filter: Filter, a: Tables.Accounts): Rep[Boolean] =
-    if (filter.accountManagers.isDefined && filter.accountManagers.get.nonEmpty) a.manager.inSet(filter.accountManagers.get) else true
-
-  private def filterAccountDelegates(filter: Filter, a: Tables.Accounts): Rep[Boolean] =
-    if (filter.accountDelegates.isDefined && filter.accountDelegates.get.nonEmpty)
-      a.delegateValue.getOrElse("").inSet(filter.accountDelegates.get)
-    else true
-
-  private def filterOperationKinds(filter: Filter, o: Tables.Operations): Rep[Boolean] =
-    if (filter.operationKinds.isDefined && filter.operationKinds.get.nonEmpty)
-      o.kind.inSet(filter.operationKinds.get) else true
-
-  private def filterOperationKindsForFees(filter: Filter, fee: Tables.Fees): Rep[Boolean] =
-    if (filter.operationKinds.isDefined && filter.operationKinds.get.nonEmpty)
-      fee.kind.inSet(filter.operationKinds.get) else true
+    (action: Query[Tables.OperationGroups, Tables.OperationGroupsRow, Seq]) extends Action
 
-  private def getFilterLimit(filter: Filter): Int = if (filter.limit.isDefined) filter.limit.get else 10
-
-  // End helper functions for constructing Slick queries
 
+  case class AccountsAction(action: Query[Tables.Accounts, Tables.AccountsRow, Seq]) extends Action
```

Comment:
More tears of joy. Can you point out the relevant slick documentation for this, because I really wish I knew this when I first wrote this code.

Reply:
I think I just looked into the code definitions and the types coming out from the queries, and I just matched those I guess.